### PR TITLE
Refactor sys/test_hooks for pytest

### PR DESCRIPTION
**Pull request by mshriver**
Tests are currently failing in master at the same point they're failing in this branch.

If that level of risk is not acceptable for the conversion, I'd like to get @rdrazny input on why this is failing on 6.9.

## Master branch test failures
```
setup@localhost:~/repos/robottelo (master$%>) % pytest -v tests/foreman/sys/test_hooks.py 
======================================================================================== test session starts =========================================================================================
platform linux -- Python 3.8.8, pytest-6.2.2, py-1.10.0, pluggy-0.13.1 -- /home/setup/repos/robottelo/.robottelo/bin/python3.8
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/setup/repos/robottelo, configfile: pyproject.toml
plugins: forked-1.3.0, ibutsu-1.14.1, xdist-2.2.1, mock-3.5.1, reportportal-5.0.8, cov-2.11.1, services-2.2.1
collected 2 items                                                                                                                                                                                    

tests/foreman/sys/test_hooks.py::ForemanHooksTestCase::test_positive_host_hooks FAILED                                                                                                         [ 50%]
tests/foreman/sys/test_hooks.py::ForemanHooksTestCase::test_positive_hostgroup_hooks FAILED                                                                                                    [100%]
====================================================================================== short test summary info =======================================================================================
FAILED tests/foreman/sys/test_hooks.py::ForemanHooksTestCase::test_positive_host_hooks - AssertionError: 1 != 0
FAILED tests/foreman/sys/test_hooks.py::ForemanHooksTestCase::test_positive_hostgroup_hooks - AssertionError: 1 != 0
============================================================================= 2 failed, 17 warnings in 72.65s (0:01:12) ==============================================================================

```

## This branch
```
setup@localhost:~/repos/robottelo (pytest-hooks-14001$%=) % pytest -v tests/foreman/sys/test_hooks.py 
========================================================================================== test session starts ==========================================================================================
platform linux -- Python 3.8.8, pytest-6.2.2, py-1.10.0, pluggy-0.13.1 -- /home/setup/repos/robottelo/.robottelo/bin/python3.8
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/setup/repos/robottelo, configfile: pyproject.toml
plugins: forked-1.3.0, ibutsu-1.14.1, xdist-2.2.1, mock-3.5.1, reportportal-5.0.8, cov-2.11.1, services-2.2.1
collected 2 items                                                                                                                                                                                       

tests/foreman/sys/test_hooks.py::test_positive_host_hooks FAILED                                                                                                                                  [ 50%]
tests/foreman/sys/test_hooks.py::test_positive_hostgroup_hooks FAILED                                                                                                                             [100%]
======================================================================================== short test summary info ========================================================================================
FAILED tests/foreman/sys/test_hooks.py::test_positive_host_hooks - assert 1 == 0
FAILED tests/foreman/sys/test_hooks.py::test_positive_hostgroup_hooks - assert 1 == 0


```

### CODE REVIEW

File: tests/foreman/sys/test_hooks.py [hunk 1]
```diff
@@ -25,137 +25,130 @@
...
+
+
+@pytest.fixture(scope='function')
+def logger_hook(default_org):
+    """Create logger script to be executed via hooks"""
+    ssh.command(
+        '''printf '#!/bin/sh\necho "$(date): Executed $1 hook'''
+        + f''' on object $2" > {LOGS_DIR}' > {SCRIPT_PATH}'''
+    )
+    ssh.command(f"chmod 774 {SCRIPT_PATH}")
```
Comment (JacobCallahan) [inline tests/foreman/sys/test_hooks.py:44]: We've still got to prefer single over double quotes

**Review by JacobCallahan — CHANGES_REQUESTED** [review]
Small changes

File: tests/foreman/sys/test_hooks.py (hunk 1 above)
Comment (mshriver) [reply tests/foreman/sys/test_hooks.py:44]: ack, this is a good time to update them. I changed them where I was modifying lines, will get the rest in this file in this branch.

**Review by JacobCallahan — APPROVED** [review]
ACK, thanks!

**Review by jameerpathan111 — APPROVED** [review]
LGTM

**Pull request merged**